src/components/Trade/hooks/useAvailableSwappers.tsx

```diff
@@ -102,9 +88,13 @@ export const useAvailableSwappers = () => {
       // Handle a race condition between form state and useTradeQuoteService
       if (
         tradeQuoteArgs?.buyAsset.assetId === buyAssetId &&
-        tradeQuoteArgs?.sellAsset.assetId === sellAssetId
+        tradeQuoteArgs?.sellAsset.assetId === sellAssetId &&
+        availableSwappersWithQuoteMetadata
       ) {
-        setSwappersWithQuoteMetadata(availableSwappersWithQuoteMetadata)
+        setSwappersWithQuoteMetadataAndTimestamp([
+          availableSwappersWithQuoteMetadata,
+          requestStartedTimestamp,
+        ])
       }
```

Comment:
could there be a risk in never setting `swappersWithQuoteMetadataAndTimestamp` to undefined such that the downstream state management tries to use them assuming different input data?

or should this be haapening?

```suggestion
      } else {
        setSwappersWithQuoteMetadataAndTimestamp(undefined)
      }
```

Reply:
You're right, we should add better protection for out-of-sync state errors in the event of network failure or other edge case. I've added a note to action this in a future PR, as it will take more propagation than the suggestion here to have an effect, and I want to keep this PR singularly.